### PR TITLE
LibJS: Avoid calling ToString on calendar when calendarName is "never"

### PR DESCRIPTION
**Pull request by Lubrsi**
This is a normative change in the Temporal spec.

See:
- https://github.com/tc39/proposal-temporal/commit/6122f4e
- https://github.com/tc39/proposal-temporal/commit/cf586bc

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp [hunk 1]
```diff
@@ -533,7 +533,21 @@ ThrowCompletionOr<PlainMonthDay*> calendar_month_day_from_fields(GlobalObject& g
     return static_cast<PlainMonthDay*>(month_day_object);
 }
 
-// 12.2.26 FormatCalendarAnnotation ( id, showCalendar ), https://tc39.es/proposal-temporal/#sec-temporal-formatcalendarannotation
+// 12.2.26 MaybeFormatCalendarAnnotation ( calendarObject, showCalendar ), https://tc39.es/proposal-temporal/#sec-temporal-maybeformatcalendarannotation
+ThrowCompletionOr<String> maybe_format_calendar_annotation(GlobalObject& global_object, Value calendar_object, StringView show_calendar)
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:537]: `calendar_object` should be an `Object&` (`Object const&`?)

**Review by linusg — CHANGES_REQUESTED** [review]

**Comment by linusg** [comment]
Ping @Lubrsi in case you forgot about this PR :)

File: Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp:537]: Actually, I was wrong about this - the spec says it must be an object, but it's called with undefined in at least one place. Will file a spec fix.

**Comment by Lubrsi** [comment]
Oh yeah, sorry, I'll rebase it now.

**Comment by linusg** [comment]
Already done :^)

**Pull request merged**